File: task-manager-api/middlewares/error_handler.py
from flask import jsonify
from werkzeug.exceptions import HTTPException


class AppError(Exception):
    def __init__(self, message, status=400):
        super().__init__(message)
        self.message = message
        self.status = status


def register_error_handlers(app):
    @app.errorhandler(AppError)
    def handle_app_error(error):
        return jsonify({"error": error.message}), error.status

    @app.errorhandler(HTTPException)
    def handle_http_error(error):
        return jsonify({"error": error.description}), error.code

    @app.errorhandler(Exception)
    def handle_unexpected(error):
        db.session.rollback()
        app.logger.exception(error)
        return jsonify({"error": "Erro interno"}), 500

    from database import db

File: ecommerce-api-legacy/src/middlewares/error-handler.js
class AppError extends Error {
    constructor(message, status = 400) {
        super(message);
        this.status = status;
    }
}

function errorHandler(error, _request, response, _next) {
    if (error instanceof AppError) {
        return response.status(error.status).json({ error: error.message });
    }
    console.error('Erro interno.', error);
    return response.status(500).json({ error: 'Erro interno' });
}

module.exports = { AppError, errorHandler };
